src/coproduct.rs

```diff
@@ -66,7 +66,7 @@ where
     /// Retrieve an effect and a sender from self if the type matches
     ///
     /// # Errors
-    /// If `self` holds an effect of a different type, this method returns an error
+    /// If `Self` holds an effect of a different type, this method returns an error
```

Comment:
I intended to refer to the run-time value and its type, so I prefer `self` over `Self` here.